### ISSUE TITLE
bug: The new ion-content > 'div.scroll-content'  breaks position:sticky and IntersectionObserver API (such as ion-virtual-scroll)

### ISSUE THREAD

**Issue by lincolnthree** [ionic-team/ionic#20258]
# Bug Report

**Ionic version:**

 **4.x**
[x] 5.x

**Current behavior:**

Any elements contained in an `ion-content` element, using `position: sticky`, will only stay stuck until the bottom of the new `.scroll-content div` implemented in issue #20185, scrolls off the screen. 

**Expected behavior:**

Sticky elements should not be pushed off the screen as `ion-content` is scrolled down. This will manifest in other ways as well (not just position-sticky), but it will also affect intersection observers (such as those used on ion-img and ion-virtual-scroll).

**Steps to reproduce:**

https://stackblitz.com/edit/ionic-v4-angular-tabs-uiajby

1. Run the sample app and scroll down on Tab1. 
2. The grey header will stick briefly, then unstick when the bottom of the .scroll-content div move up off the screen.

**Related code:**
````
<ion-content>
    <ion-card class="welcome-card">
      <ion-card-header>
        <ion-card-subtitle>Get Started</ion-card-subtitle>
        <ion-card-title>Welcome to Ionic</ion-card-title>
      </ion-card-header>
      <ion-card-content>
        <p>Now that your app has been created, you'll want to start building out features and components. Check out some of the resources below for next steps.</p>
      </ion-card-content>
    </ion-card>


    <!-- This element will scroll off the viewport once the bottom of the ion-content .scroll-content div is reached -->
    <ion-list-header class="sticky-header">
      <ion-label class=ion-text-wrap>Resources (Scroll down to stick this element and see viewport issue)</ion-label>
    </ion-list-header>

````
https://stackblitz.com/edit/ionic-v4-angular-tabs-uiajby

```
.sticky-header {
  position: sticky;
  top: 0;
  z-index: 100000;
  background-color: grey;
}
```

**Other information:**

This is likely fixed by using the CSS `min-height: 100%` attribute on the .scroll-content div, rather than the `height: 100%` attribute.

Shadow DOM is great but makes it very difficult to work around issues like this. Not sure what the compromise needs to be, however, as no layout will be perfect.

**Ionic info:**

```
Ionic:

   Ionic CLI                     : 5.4.13 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 5.0.0-beta.5
   @angular-devkit/build-angular : 0.900.0-rc.9
   @angular-devkit/schematics    : 9.0.0-rc.9
   @angular/cli                  : 9.0.0-rc.9
   @ionic/angular-toolkit        : 2.1.1

System:

   Android SDK Tools : 26.1.1 (/Users/lincoln/Library/Android/sdk)
   NodeJS            : v12.13.0 (/usr/local/lib/node_modules/node/bin/node)
   npm               : 6.13.6
   OS                : macOS Catalina
   Xcode             : Xcode 11.3.1 Build version 11C504

```

**Comment by lincolnthree** [comment]
![giphy](https://user-images.githubusercontent.com/362329/72849049-e63ec380-3c73-11ea-87c3-897e0b2104cf.gif)

**Comment by lincolnthree** [comment]
This also breaks ````<ion-virtual-scroll>````

**Comment by lincolnthree** [comment]
@liamdebeasi Just FYI: when you get to this one, I did a quick hack of the DOM and confirmed that 'min-height: 100%; height: auto;' on `ion-content > :shadow > slot > div.scroll-content` fixes the issue as it allows the 'overflowed' content to expand the height of the scroll div.

**Comment by liamdebeasi** [comment]
Can you give this dev build a try? 

`npm i @ionic/angular@5.0.0-dev.202001221425.f896821`

**Comment by lincolnthree** [comment]
@liamdebeasi Looks like that takes care of the issue. Thanks for the quick patch!

**Comment by liamdebeasi** [comment]
Glad to hear it's working. I've fixed the issue via #20258, and the fix will be available in an upcoming release of Ionic Framework.
